pytube/__main__.py

```diff
@@ -166,6 +168,7 @@ def prefetch(self) -> None:
         self.watch_html = request.get(url=self.watch_url)
         if self.watch_html is None:
             raise VideoUnavailable(video_id=self.video_id)
+        self.publish_date = extract.publish_date(self.watch_html)
```

Comment:
Instead of assigning the publish_date variable, this should probably be an `@property` of the class

Reply:
Good point. Because not all attributes were implemented as "property" I implemented it like this. Will be changed.